### PR TITLE
Failed to Lock when fetching lib

### PR DESCRIPTION
**Pull request by MartinNowak**
Windows 10
Dub: 1.3.0

cmd: dub run --build=debug --arch=x86 --compiler=dmd

```
Fetching libevent 2.0.2+2.0.16 (getting selected version)...
Failed to lock 'C:\Users\User Name\AppData\Roaming\dub\packages\libevent-2.0.2_2.0.16\libevent.lock'.

on dub.json:
"dependencies": {
     "requests": "~>0.3.2",
     "htmld": "~>0.2.16"
}
```

Maybe is relate to long path names?

### CODE REVIEW

File: source/dub/internal/utils.d [hunk 1]
```diff
@@ -45,32 +45,42 @@ Path getTempFile(string prefix, string extension = null)
...
+     The locked file or an Exception on timeout.
+*/
+auto lockFile(string path, Duration timeout)
 {
-	bool opCast(T:bool)() { return !!path; }
-	~this() { if (path) rmdir(path); }
-	string path;
-}
+	import std.datetime, std.stdio : File;
+	import std.algorithm.mutation : move;
```
Comment (s-ludwig) [inline source/dub/internal/utils.d]: Fails on 2.066 and below -> `std.algorithm : move`

**Comment by s-ludwig** [comment]
Looks good. I can also confirm that I ran into this a few times when I cancelled a package download using Ctrl+C.

**Comment by MartinNowak** [comment]
Yikes, stupid https://github.com/dlang-bots/dlang-bot/issues/69, hope the tests will be all right.

**Comment by wilzbach** [comment]
Yeah seems like stable didn't had required CI checks.
I just enabled them for now.

image

**Pull request merged**